modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceDiscounts.testcase

```diff
@@ -227,6 +227,163 @@ definition {
 			value1 = "24.00");
 	}
 
+	test CanViewErrorNotificationWhenApplyingCouponCodeToWrongSku {
+		property custom.properties = "jsonws.web.service.paths.excludes=${line.separator}company.security.strangers.verify=false";
+		property portal.acceptance = "false";
+
+		task ("Given a Minium site with a buyer user") {
+			CommerceAccelerators.initializeNewSiteViaAccelerator(siteName = "Minium");
+
+			CommerceEntry.addAccountEntryUser(
+				accountName = "Commerce Account",
+				accountType = "Business",
+				agreeToTermsAndAnswerReminderQuery = "true",
+				createNewAccount = "true",
+				requireReset = "false",
+				userEmailAddress = "buyer@liferay.com",
+				userFirstName = "Buyer",
+				userLastName = "User",
+				userRole = "Buyer",
+				userScreenName = "buyeruser",
+				userSiteMembership = "Minium");
+		}
+
+		task ("And a discount of 10% applied to ABS Sensor's SKU") {
+			CommerceNavigator.gotoPortlet(
+				category = "Pricing",
+				portlet = "Discounts");
+
+			Click(locator1 = "Button#PLUS");
+
+			CommerceEntry.addNewDiscountsEntry(
+				discountAmount = "10",
+				discountName = "Test Discount",
+				discountTarget = "Products",
+				discountType = "Percentage");
+
+			CommerceRelations.assignSingleRelationToProductsEntry(
+				entryName = "ABS Sensor",
+				title = "Add Products");
+		}
+
+		task ("And a coupon code added for the discount") {
+			Click(
+				key_name = "Coupon Code",
+				locator1 = "ToggleSwitch#ENABLE_SPECIFIC_TOGGLE_SWITCH");
+
+			Type(
+				key_fieldLabel = "Coupon Code",
+				locator1 = "TextInput#GENERIC_TEXT_INPUT",
+				value1 = "TEST");
+
+			Button.clickPublish();
+		}
+
+		task ("And a coupon code entry widget added to the checkout page") {
+			Navigator.openWithAppendToBaseURL(urlAppend = "group/minium/checkout");
+
+			Click(locator1 = "Icon#PLUS");
+
+			Type(
+				key_text = "Search",
+				locator1 = "TextInput#ANY",
+				value1 = "Coupon Code Entry");
+
+			MouseOver(
+				categoryName = "Coupon Code Entry",
+				locator1 = "ContentDashboard#CATEGORY_NAME_ELEMENT");
+
+			Click(locator1 = "Button#PLUS");
```

Comment:
Try to use `CommerceAcceleratorsInitializers.addPortlet()` instead of this and see if it works. Thanks!